src/main/java/org/swisspush/redisques/util/RedisquesConfiguration.java

```diff
@@ -62,7 +64,7 @@ public class RedisquesConfiguration {
     // - https://groups.google.com/g/vertx/c/fe0RSWEfe8g
     // - https://vertx.io/docs/apidocs/io/vertx/redis/client/RedisOptions.html#setMaxPoolWaiting-int-
     // - https://stackoverflow.com/questions/59692663/vertx-java-httpclient-how-to-derive-maxpoolsize-and-maxwaitqueuesize-values-and
-    private static final int DEFAULT_REDIS_MAX_POOL_WAIT_SIZE = -1;
+    private static final int DEFAULT_REDIS_MAX_POOL_WAIT_SIZE = Integer.MAX_VALUE;
```

Comment:
The cluster client wants the value to be larger compared to the pool size

  public RedisClusterClient(Vertx vertx, RedisOptions options) {
    super(vertx, options);
    this.options = options;
    // validate options
    if (options.getMaxPoolWaiting() < options.getMaxPoolSize()) {
      throw new IllegalStateException("Invalid options: maxPoolWaiting < maxPoolSize");


-1 means unlimited, now we make it veryvery large.

The value is used in the SimpleConnectionPool below. I would say for our case things remain the same, we do not allocate extra resources and the number of waiters cannot go beyond Integer.MAX_VALUE anyway (because data type int is used).

      if (pool.maxWaiters == -1 || (pool.waiters.size() + pool.requests) < pool.maxWaiters) {
        pool.waiters.addLast(this);